### DISCUSSION TITLE
Web Rotation in Bar Elements

### DISCUSSION THREAD

**Discussion by jose-ortolano** [BriefFiniteElementNet/BriefFiniteElement.Net#160] [Q&A] [answered]
Hi!

I'm doing some tests with Brief Finite Element.NET. Currently, I'm trying to validate the bar model results with a few simple examples.

I was trying to calculate a bar with the cross section rotated 45º, using the "WebRotation" property. However, I am not getting the expected results: it doesn't seem to have an effect on my bar deformations.

Any information or help will be appreciated. Thank you in advance!

Kind Regards,

**Comment by epsi1on** [comment]
Hello
Can you please write the code here that reproduces this situation?

Thanks

**Reply by jose-ortolano** [reply]
[BeamValidation.txt](https://github.com/BriefFiniteElementNet/BriefFiniteElement.Net/files/12282275/BeamValidation.txt)

Please finde the code attached. It is a beam with a rectangular cross-section. I am trying to analyze the deformation when I rotated the bar 45º.

Thank you!

**Reply by epsi1on** [reply]
Thanks. 
Your code consist of some other parts too. A little large and hard to follow/understand.
Can you confirm that, on a simple model like cantilever beam (fixed at one end and free on other end) with rectangular section, having web rotation is effect less on the result?
If yes, I can check that...

**Reply by jose-ortolano** [reply]
Hi again,

I have tried again with a cantilever beam and web rotation has no effect on results. These are my input values:

- Material: Steel (E = 2.1e5 MPa, Poisson=0.30)
- Cross section: rectangular (Height: 200 mm, Width: 10 mm).
- Beam Length = 10 meters.
- Uniform Load: 1KN/m acting on global Z-Axis.

Using this values the program gives me a maximum deflection equal to 892,9 mm, which is correct when web rotation is 0º. This value does not change when I calculated using web rotation equal to 45º.

Thank you again.

**Reply by epsi1on** [reply]
I confirm there where a bug that causes the `BarElement.WebRotation` be neglected. It is fixed now.
here is test code
https://github.com/BriefFiniteElementNet/BriefFiniteElement.Net/blob/master/Validation/Validation/GithubIssues/Issue159.cs

**Reply by jose-ortolano** [reply]
Thank you again for the quick answer!

Is it planned to have a fixed version as NugetPackage? That would be great. 

Thank you again! The tool is great, congrats!

**Reply by epsi1on** [reply]
Yes, Source is updated, so it will be reflected in next nuget update.

**Reply by epsi1on** [reply]
Packages are updated

**Reply by jose-ortolano** [reply, marked as answer]
Thank you very much!!! And again, congrats for the amazimg tool!!

**Reply by epsi1on** [reply]
You are welcome. :) Thanks for reporting the bug.
I'll mark this as answer in order to close the discussion...

**Comment by epsi1on** [comment]
inactivity
